src/private_API.jl

```diff
@@ -0,0 +1,124 @@
+# This file contains all functions that either do not directly interface with `brew`
+# or peek into the internals of Homebrew a bit, such as `install_brew()` which
+# installs Homebrew using Git, `tap_exists()` which presupposes knowledge of the
+# internal layout of Taps, and `installed()` which presupposed knowledge of the
+# location of files within the Cellar
+
+const tappath = joinpath(brew_prefix,"Library","Taps","staticfloat","homebrew-juliadeps")
+const BREW_URL = "https://github.com/Homebrew/brew"
+const BREW_BRANCH = "master"
+const BOTTLE_SERVER = "https://juliabottles.s3.amazonaws.com"
+
+
+"""
+install_brew()
+
+Ensures that Homebrew is installed as desired, that our basic Taps are available
+and that we have whatever binary tools we need, such as `install_name_tool`
+"""
+function install_brew()
+    # Ensure brew_prefix exists
+    if !isdir(brew_prefix)
+        mkdir(brew_prefix)
+
+        try
+            Base.info("Downloading brew...")
+            @compat run(pipeline(`curl -# -L $BREW_URL/tarball/$BREW_BRANCH`,
+                                 `tar xz -m --strip 1 -C $brew_prefix`))
+        catch
+            warn("Could not download/extract $BREW_URL/tarball/$BREW_BRANCH into $(brew_prefix)!")
+            rethrow()
+        end
+    end
+
+    # Tap homebrew/core, always and forever
+    tap("homebrew/core")
+
+    # Tap our own "overrides" tap
+    tap("staticfloat/juliadeps")
+
+    if !clt_installed()
+        # If we don't have the command-line tools installed, then let's grab
+        # cctools, and `git` if we don't already have them, as they're important
+        if !installed("cctools")
+            add("cctools")
```

Comment:
Yes. There's gonna be a lot of that over the next few days. I'm going to be
MIA for the next 24 hours or so, so if you want to merge, go ahead, but I
won't be available for questions for a bit.
-E

On Sun, Jun 5, 2016, 22:19 Tony Kelman <notifications@github.com> wrote:

> In src/private_API.jl
> <https://github.com/JuliaLang/Homebrew.jl/pull/113#discussion_r65838119>:
>
> > +            warn("Could not download/extract $BREW_URL/tarball/$BREW_BRANCH into $(brew_prefix)!")
> > +            rethrow()
> > +        end
> > +    end
> > +
> > +    # Tap homebrew/core, always and forever
> > +    tap("homebrew/core")
> > +
> > +    # Tap our own "overrides" tap
> > +    tap("staticfloat/juliadeps")
> > +
> > +    if !clt_installed()
> > +        # If we don't have the command-line tools installed, then let's grab
> > +        # cctools, and `git` if we don't already have them, as they're important
> > +        if !installed("cctools")
> > +            add("cctools")
>
> so this is an upstream formula now?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/JuliaLang/Homebrew.jl/pull/113/files/e0e030bdb88cc9c765b40251910b226ebc6d94fd#r65838119>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAH_aLkRLeTX95daD7iMjTareh_D1ESIks5qI63jgaJpZM4IujLy>
> .
>
-- 

-E